lib/spack/spack/spec.py

```diff
@@ -1004,6 +1004,10 @@ def __init__(self, spec_like=None,
         self.external_module = external_module
         self._full_hash = full_hash
 
+        # This attribute is used to store custom information for
+        # external specs. None signal that it was not set yet.
+        self.extra_attributes = None
```

Comment:
I think Spack-installed instances of gcc would want to store extra information as well.

Reply:
I was more thinking that if an instance is Spack installed, then these extra attributes (or any other property exposed by a class that depends indirectly on them) could be computed e.g. by inspecting the installation prefix. As an example see https://github.com/spack/spack/pull/16526#pullrequestreview-413535582 where:
```python
spec.package.cc
```
is computed either from an extra attribute or from the prefix.

Reply:
This makes sense and may not work the way I suggested it (for example your method works if the package is relocated). Thanks for the pointer.